meson.build

```diff
@@ -0,0 +1,53 @@
+project(
+  'scikit-learn',
+  'c', 'cpp', 'cython',
+  version: '1.5.0.dev0',
```

Comment:
How do we deal with having the version in multiple places (here, `sklearn/__init__.py`)? 

In the olden days one end of the spectrum is having to keep it in sync by hand which is annoying but simple. At the other end of the spectrum was something like versioneer, which is convenient but quite complex. It would be nice to find a solution that is somewhere in between/maybe a bit on the side of simple.

Reply:
I have not done it yet (very courageously leaving it for later) but I think scikit-image (and probably others) have something that reads the root `__init__.py` see https://github.com/scikit-image/scikit-image/blob/main/meson.build#L7

Reply:
You can use a meson configure_file() to take the version number from meson.build and edit it into e.g. `_version.py.in` (probably easier than `__init__.py`) which will get installed.

Alternatively, you could use a meson `project(... version: run_command('get_version.py').stdout().strip()` using a script that processes your `__init__.py` to extract the version number and import it into the meson definition. If you do this, the `ast` module can be helpful to avoid needing to import code that has not been compiled yet.